pkg/engine2/operational_rule/operational_step.go

```diff
@@ -306,6 +316,11 @@ func (ctx OperationalRuleContext) SetField(resource, fieldResource *construct.Re
 		return fmt.Errorf("error appending field %s#%s with %s: %w", resource.ID, path, fieldResource.ID, err)
 	}
 	zap.S().Infof("appended field %s#%s with %s", resource.ID, path, fieldResource.ID)
+	ctx.namespace(resource, fieldResource, oldId)
+	return nil
+}
+
+func (ctx OperationalRuleContext) namespace(resource, fieldResource *construct.Resource, oldId construct.ResourceId) {
```

Comment:
```suggestion
func (ctx *OperationalRuleContext) namespace(resource, fieldResource *construct.Resource, oldId construct.ResourceId) {
```
nit: this works as-is because `.Data` is a pointer so it gets copied over, but for methods that mutate I generally prefer them clearly marked such by using a pointer receiver.